src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -180,6 +180,13 @@ public int getTotalMessageCount(){
     return results.countEntities(FetchOptions.Builder.withDefaults());
   }
 
+  public int getUserMessageCount(String email) {
```

Comment:
I think this is an perfectly fine option, but just offering an alternative:

When counting TotalMessage counts, break down the count by each user, so that we have the total # and per user count with a single call, instead of N + 1 calls for N users. 

The trade off is that this will cause a) some duplicate logic since we still need getUserMessageCount for per-user stats, or b) use the modified getTotalMessageCount for per-user stats too, at the cost of possibly more wasteful query.

Reply:
Yeah this is something I definitely planned on looking into, it can be done in a single call. I'll try fix it before merging, thanks for pointing it out!

Reply:
Made a change so that only one call is made, I left the old method there because it may be useful in the future